icu4j/main/core/src/main/java/com/ibm/icu/util/LocaleData.java

```diff
@@ -421,6 +426,9 @@ public static final PaperSize getPaperSize(ULocale locale){
      * @stable ICU 4.2
      */
     public String getLocaleDisplayPattern() {
+        if (langBundle == null) {
+            throw new ICUException("Language data missing for getLocaleDisplayPattern");
```

Comment:
UResourceBundle.getBundleInstance() would have thrown a MissingResourceException, right? As in C++, I think throwing that would be more appropriate. Or even storing the real exception but maybe that's overkill?